runtime/src/commitment.rs

```diff
@@ -215,6 +216,61 @@ impl CommitmentSlots {
     }
 }
 
+#[derive(Default, Clone)]
+pub struct CommitmentBanks {
+    /// The slot of the bank from which all other slots were calculated.
+    pub bank: Arc<Bank>,
+    /// The current node root
+    pub root: Option<Arc<Bank>>,
+    /// Highest cluster-confirmed slot
+    pub highest_confirmed_slot: Option<Arc<Bank>>,
+    /// Highest cluster-confirmed root
+    pub highest_confirmed_root: Option<Arc<Bank>>,
+}
+
+impl CommitmentBanks {
+    pub fn new_from_bank(bank: Arc<Bank>) -> Self {
+        Self {
+            bank,
+            ..Self::default()
+        }
+    }
+
+    pub fn commitment_slots(&self) -> CommitmentSlots {
+        CommitmentSlots {
+            slot: self.slot(),
+            root: self.root_slot(),
+            highest_confirmed_slot: self.highest_confirmed_slot(),
+            highest_confirmed_root: self.highest_confirmed_root_slot(),
+        }
+    }
+
+    pub fn slot(&self) -> Slot {
+        self.bank.slot()
+    }
+
+    pub fn root_slot(&self) -> Slot {
```

Comment:
These methods are just here to ease the transition. Using it will oftentimes leave the caller in the form `let bank = bank_forks.get(commitment_banks.root_slot());` which could be trivially reduced to `let bank = commitment_banks.root;`, which removes the BankForks dependency.